src/librustc_typeck/coherence/overlap.rs

```diff
@@ -65,6 +65,23 @@ impl<'cx, 'tcx> OverlapChecker<'cx, 'tcx> {
                 continue;
             }
 
+            if self.tcx.default_trait_impls.borrow().contains(&trait_def_id) {
+                match ty::trait_impl_polarity(self.tcx, impl1_def_id) {
+                    Some(ast::ImplPolarity::Positive) => {
+                        // It's possible to opt-out from a default trait implementation
+                        // but not overlap with it. Therfore, a positive trait implementation
+                        // is considered an overlap over an existing default implementation.
```

Comment:
I'm not sure I follow this logic. It seems to be saying that if you have a default impl, you cannot have a positive impl? (But the comment says *negative impl*?) Both seem wrong to me, though.